drivers/display/display_st7701.c

```diff
@@ -0,0 +1,459 @@
+	if (ret < 0) {
+		return ret;
+	}
+
+	buf[0] = 0x00;
+	buf[1] = 0x00;
+	sys_put_be16(data->xres, (uint8_t *)&buf[2]);
+	ret = st7701_dcs_write(dev, MIPI_DCS_SET_COLUMN_ADDRESS, buf, 4);
+	if (ret < 0) {
+		return ret;
+	}
+
+	buf[0] = 0x00;
+	buf[1] = 0x00;
+	sys_put_be16(data->yres, (uint8_t *)&buf[2]);
+	ret = st7701_dcs_write(dev, MIPI_DCS_SET_PAGE_ADDRESS, buf, 4);
+	if (ret < 0) {
+		return ret;
+	}
+
+	/* Backlight control */
+	buf[0] = ST7701_WRCTRLD_BCTRL | ST7701_WRCTRLD_DD | ST7701_WRCTRLD_BL;
+	ret = st7701_dcs_write(dev, MIPI_DCS_WRITE_CONTROL_DISPLAY, buf, 1);
+	if (ret < 0) {
+		return ret;
+	}
+
+	/* Adaptive brightness control */
+	buf[0] = ST7701_WRCABC_UI;
+	ret = st7701_dcs_write(dev, MIPI_DCS_WRITE_POWER_SAVE, buf, 1);
+	if (ret < 0) {
+		return ret;
+	}
+
+	/* Adaptive brightness control minimum brightness */
+	buf[0] = 0xFF;
+	ret = st7701_dcs_write(dev, MIPI_DCS_SET_CABC_MIN_BRIGHTNESS, buf, 1);
+	if (ret < 0) {
+		return ret;
+	}
+
+	/* Brightness */
+	buf[0] = 0xFF;
+	ret = st7701_dcs_write(dev, MIPI_DCS_SET_DISPLAY_BRIGHTNESS, buf, 1);
+	if (ret < 0) {
+		return ret;
+	}
+
+	/* Display On */
+	ret = st7701_dcs_write(dev, MIPI_DCS_SET_DISPLAY_ON, NULL, 0);
+	if (ret < 0) {
+		return ret;
+	}
+
+	return 0;
+}
+
+static int st7701_blanking_on(const struct device *dev)
+{
+	const struct st7701_config *cfg = dev->config;
+	int ret;
+
+	if (cfg->backlight.port != NULL) {
+		ret = gpio_pin_set_dt(&cfg->backlight, 0);
+		if (ret) {
+			LOG_ERR("Disable backlight failed! (%d)", ret);
+			return ret;
+		}
+	}
+
+	return st7701_dcs_write(dev, MIPI_DCS_SET_DISPLAY_OFF, NULL, 0);
+}
+
+static int st7701_blanking_off(const struct device *dev)
+{
+	const struct st7701_config *cfg = dev->config;
+	int ret;
+
+	if (cfg->backlight.port != NULL) {
+		ret = gpio_pin_set_dt(&cfg->backlight, 1);
+		if (ret) {
+			LOG_ERR("Enable backlight failed! (%d)", ret);
+			return ret;
+		}
+	}
+
+	return st7701_dcs_write(dev, MIPI_DCS_SET_DISPLAY_ON, NULL, 0);
+}
+
+static int st7701_write(const struct device *dev, uint16_t x, uint16_t y,
+			const struct display_buffer_descriptor *desc, const void *buf)
+{
+	return -ENOTSUP;
+}
+
+static int st7701_set_brightness(const struct device *dev, uint8_t brightness)
+{
+	return st7701_dcs_write(dev, MIPI_DCS_SET_DISPLAY_BRIGHTNESS, &brightness, 1);
+}
+
+static void st7701_get_capabilities(const struct device *dev,
+				struct display_capabilities *capabilities)
+{
+	const struct st7701_config *cfg = dev->config;
+	struct st7701_data *data = dev->data;
+
+	if (!capabilities) {
+		return -EINVAL;
+	}
+
+	memset(capabilities, 0, sizeof(struct display_capabilities));
+	capabilities->x_resolution = cfg->width;
+	capabilities->y_resolution = cfg->height;
+	capabilities->supported_pixel_formats = data->pixel_format;
+	capabilities->current_pixel_format = data->pixel_format;
+	capabilities->current_orientation = data->orientation;
+}
+
+static const struct display_driver_api st7701_api = {
```

Comment:
Wrap the device API with the device API macro

```suggestion
static DEVICE_API(display, st7701_api) = {
```